### PR TITLE
Moved key table type to Row, and misc performance improvements

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
Any comments?

File: src/main/scala/is/hail/keytable/KeyTable.scala [hunk 1]
```diff
@@ -127,6 +125,16 @@ case class KeyTable(hc: HailContext, rdd: RDD[Annotation],
...
 
+  def keySignature: TStruct = {
+    val keySet = keyNames.toSet
+    TStruct(signature.fields.filter(f => keySet.contains(f.name)).map(f => (f.name, f.typ)): _*)
+  }
+
+  def valueSignature: TStruct = {
+    val keySet = keyNames.toSet
+    TStruct(signature.fields.filter(f => !keySet.contains(f.name)).map(f => (f.name, f.typ)): _*)
+  }
```
Comment (jigold) [inline src/main/scala/is/hail/keytable/KeyTable.scala]: The `keySignature` and `valueSignature` aren't used anywhere.

File: src/main/scala/is/hail/keytable/KeyTable.scala [hunk 2]
```diff
@@ -359,17 +362,21 @@ case class KeyTable(hc: HailContext, rdd: RDD[Annotation],
...
+
+      if (r2 != null) {
         mergeIndices.zipWithIndex.foreach { case (i, j) => result(size1 + j) = r2(i) }
       }
-      Annotation.fromSeq(result)
+      if (r2 != null) {
+        mergeIndices.indices.foreach { i =>
+          result(size1 + i) = r2(mergeIndices(i))
+        }
+      }
```
Comment (jigold) [inline src/main/scala/is/hail/keytable/KeyTable.scala]: This is repeated directly above.

**Comment by tpoterba** [comment]
Back to you. Also, valueSignature and keySignature are now used -- I changed keyedRDD to not duplicate keys and values

**Comment by tpoterba** [comment]
back to you

**Pull request merged**